src/System.Net.Http/src/System/Net/Http/HttpContent.cs

```diff
@@ -202,7 +202,17 @@ internal static string ReadBufferAsString(ArraySegment<byte> buffer, HttpContent
             {
                 try
                 {
-                    encoding = Encoding.GetEncoding(headers.ContentType.CharSet);
+                    // Remove at most a single set of quotes.
+                    if (headers.ContentType.CharSet.Length > 2 &&
```

Comment:
What would be the expected result when parsing `charset=""`?

Reply:
The current behavior (not modified by this change) is to throw `InvalidOperationException`. Our parser treats it as a legal Content-Type header but an unknown header, so it falls under case (3) in my comment above.